src/test/java/uk/co/ramp/policy/alert/AlertContactTracerTest.java

```diff
@@ -53,12 +53,29 @@ public void testPrimaryLevelTrace() {
         .thenReturn(List.of(event1, event2, event3));
     when(population.getAlertStatus(anyInt())).thenReturn(AlertStatus.NONE);
     when(population.getVirusStatus(anyInt())).thenReturn(VirusStatus.EXPOSED);
+    when(population.hasApp(anyInt())).thenReturn(true);
+
     var contactTracer =
         new AlertContactTracer(tracingPolicy, eventList, population, statisticsRecorder);
 
     assertThat(contactTracer.traceRecentContacts(0, 2, 1)).containsExactlyInAnyOrder(2, 3);
   }
 
+  @Test
+  public void testPrimaryLevelTraceNoApp() {
```

Comment:
This includes a lot of copy-and-pasted code from the previous test.  Could these tests be refactored to share common code?

Reply:
refactored